### PR TITLE
rescue: Initial version of the rescue disk

### PR DESCRIPTION
**Pull request by andr2000**
Add ability to build an initramfs based image
suitable for booting on Salvator-X M3/H3 with full
hardware support and required set of maintenance utilities
such as file utilities, package management (dpkg, rpm, opkg)
suitable for system recovery and tuning.

Signed-off-by: Oleksandr Andrushchenko <oleksandr_andrushchenko@epam.com>

### CODE REVIEW

File: recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/doc/local.conf.dom0-image-rescue-initramfs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+# the option is a question of removing the # character and making any change to the
+# variable as required.
+
+#
+# Machine Selection
+#
+# You need to select a specific machine to target the build with. There are a selection
+# of emulated machines available which can boot and run in the QEMU emulator:
+#
+MACHINE = "generic-armv8-xt"
```
Comment (lorc) [inline recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/doc/local.conf.dom0-image-rescue-initramfs:20]: I'm not sure that `generic-armv8-xt` machine is able to work with renesas peripherals.

File: recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig [hunk 2]
```diff
@@ -0,0 +1,5699 @@
...
+CONFIG_ARCH_SUPPORTS_ATOMIC_RMW=y
+CONFIG_MUTEX_SPIN_ON_OWNER=y
+CONFIG_RWSEM_SPIN_ON_OWNER=y
+CONFIG_LOCK_SPIN_ON_OWNER=y
+CONFIG_FREEZER=y
+
+#
+# Platform selection
+#
+CONFIG_ARCH_SUNXI=y
```
Comment (lorc) [inline recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig:326]: Do we need all those platforms?

File: recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig [hunk 3]
```diff
@@ -0,0 +1,5699 @@
...
+CONFIG_PCI_IOV=y
+# CONFIG_PCI_PRI is not set
+# CONFIG_PCI_PASID is not set
+CONFIG_PCI_LABEL=y
+# CONFIG_HOTPLUG_PCI is not set
+
+#
+# PCI host controller drivers
+#
+CONFIG_PCI_AARDVARK=y
```
Comment (lorc) [inline recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig:393]: Do we need to support all this PCI HCs?

File: recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig [hunk 4]
```diff
@@ -0,0 +1,5699 @@
...
+CONFIG_KEXEC=y
+CONFIG_XEN_DOM0=y
+CONFIG_XEN=y
+CONFIG_FORCE_MAX_ZONEORDER=11
+# CONFIG_ARMV8_DEPRECATED is not set
+
+#
+# ARMv8.1 architectural features
+#
+CONFIG_ARM64_HW_AFDBM=y
```
Comment (lorc) [inline recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig:512]: R3/H3 are built on ARMv8.0 arch

File: recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig [hunk 5]
```diff
@@ -0,0 +1,5699 @@
...
+CONFIG_ARM64_PAN=y
+# CONFIG_ARM64_LSE_ATOMICS is not set
+CONFIG_ARM64_VHE=y
+
+#
+# ARMv8.2 architectural features
+#
+CONFIG_ARM64_UAO=y
```
Comment (lorc) [inline recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig:520]: R3/H3 are built on ARMv8.0 arch

File: recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig [hunk 6]
```diff
@@ -0,0 +1,5699 @@
...
+CONFIG_ARM64_MODULE_CMODEL_LARGE=y
+# CONFIG_RANDOMIZE_BASE is not set
+
+#
+# Boot options
+#
+# CONFIG_ARM64_ACPI_PARKING_PROTOCOL is not set
+CONFIG_CMDLINE=""
+# CONFIG_CMDLINE_FORCE is not set
+CONFIG_EFI_STUB=y
```
Comment (lorc) [inline recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig:530]: EFI is not supported on M3/H3

File: recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig [hunk 7]
```diff
@@ -0,0 +1,5699 @@
...
+# CPU frequency scaling drivers
+#
+CONFIG_CPUFREQ_DT=y
+CONFIG_CPUFREQ_DT_PLATDEV=y
+CONFIG_ARM_BIG_LITTLE_CPUFREQ=y
+# CONFIG_ARM_DT_BL_CPUFREQ is not set
+# CONFIG_ARM_KIRKWOOD_CPUFREQ is not set
+# CONFIG_ARM_MT8173_CPUFREQ is not set
+CONFIG_ARM_SCPI_CPUFREQ=y
+CONFIG_ARM_TEGRA20_CPUFREQ=y
```
Comment (lorc) [inline recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig:618]: M3/H3 is not  Tegra SoC.

File: recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig [hunk 8]
```diff
@@ -0,0 +1,5699 @@
...
+#
+# CONFIG_ARM_CCI400_PMU is not set
+# CONFIG_ARM_CCI5xx_PMU is not set
+# CONFIG_ARM_CCN is not set
+# CONFIG_QCOM_EBI2 is not set
+# CONFIG_SIMPLE_PM_BUS is not set
+# CONFIG_SUNXI_RSB is not set
+# CONFIG_TEGRA_ACONNECT is not set
+CONFIG_UNIPHIER_SYSTEM_BUS=y
+CONFIG_VEXPRESS_CONFIG=y
```
Comment (lorc) [inline recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig:1077]: ... and is not VExpress also

File: recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig [hunk 9]
```diff
@@ -0,0 +1,5699 @@
...
+# CONFIG_SCSI_AM53C974 is not set
+# CONFIG_SCSI_WD719X is not set
+# CONFIG_SCSI_DEBUG is not set
+# CONFIG_SCSI_PMCRAID is not set
+# CONFIG_SCSI_PM8001 is not set
+# CONFIG_SCSI_VIRTIO is not set
+# CONFIG_SCSI_LOWLEVEL_PCMCIA is not set
+# CONFIG_SCSI_DH is not set
+# CONFIG_SCSI_OSD_INITIATOR is not set
+CONFIG_HAVE_PATA_PLATFORM=y
```
Comment (lorc) [inline recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig:1377]: No PATA in Salvator boards.

File: recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig [hunk 10]
```diff
@@ -0,0 +1,5699 @@
...
+# CONFIG_NETXEN_NIC is not set
+# CONFIG_QED is not set
+CONFIG_NET_VENDOR_QUALCOMM=y
+# CONFIG_QCA7000 is not set
+# CONFIG_QCOM_EMAC is not set
+CONFIG_NET_VENDOR_REALTEK=y
+# CONFIG_8139CP is not set
+# CONFIG_8139TOO is not set
+# CONFIG_R8169 is not set
+CONFIG_NET_VENDOR_RENESAS=y
```
Comment (lorc) [inline recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/recipes-kernel/linux/linux-generic-armv8/defconfig:1668]: We need only `CONFIG_NET_VENDOR_RENESAS`

File: recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/doc/local.conf.dom0-image-rescue-initramfs (hunk 1 above)
Comment (andr2000) [reply recipes-dom0/dom0-image-rescue-initramfs/files/meta-xt-prod-extra/doc/local.conf.dom0-image-rescue-initramfs:20]: well, in order to see proper machine names here we have to pull in meta-rcar-gen3+
at the same time, the root filesystem is really generic as it doesn't contain anything R-Car specific, but only kernel is

**Comment by andr2000** [comment]
once on all defconfig: this is from prod-devel as is. of course there is room for optimization

**Comment by lorc** [comment]
`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

**Comment by andr2000** [comment]
@iartemenko pls merge

**Pull request merged**